front/lib/data_sources.ts

```diff
@@ -33,3 +38,39 @@ export function getDisplayNameForDataSource(ds: DataSourceType) {
     return ds.name;
   }
 }
+
+type DataSource = DataSourceType | DataSourceResource;
+
+export function isFolder(
+  ds: DataSource
+): ds is DataSource & { connectorProvider: null } {
+  // If there is no connectorProvider, it's a folder.
+  return !ds.connectorProvider;
+}
+
+export function isWebsite(
+  ds: DataSource
+): ds is DataSource & { connectorProvider: "webcrawler" } {
+  return ds.connectorProvider === "webcrawler";
+}
+
+export function isManaged(ds: DataSource): ds is DataSource & {
+  connectorProvider: ConnectorProvider;
+  connectorId: string;
+} {
```

Comment:
`nit`
We could have created a type `WithConnector` to apply `DataSourceType` to avoid duplicating this type 😉